lib/measured/measurable.rb

```diff
@@ -6,7 +6,7 @@ class Measured::Measurable < Numeric
   def initialize(value, unit)
     raise Measured::UnitError, "Unit value cannot be blank" if value.blank?
 
-    @unit = self.class.unit_system.to_unit_name!(unit)
+    @unit = unit.is_a?(Measured::Unit) ? unit : self.class.unit_system.unit_for!(unit)
```

Comment:
This could be extracted into a `unit_from_unit_or_name(arg)` method which would pass through `Unit` or look it up. Then use it again on 23.